### PR TITLE
Upgrade notes to elasticsearch7

### PR DESCRIPTION
**Pull request by stvstnfrd**
Note: This depends on https://github.com/edx/devstack/pull/612 .
This branch will be rebased and the PR re-pointed to `master` following #612 merging to `master`.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -264,7 +264,7 @@ services:
       DB_USER: "notes001"
       DJANGO_WATCHMAN_TIMEOUT: 30
       ENABLE_DJANGO_TOOLBAR: 1
-      ELASTICSEARCH_URL: "http://edx.devstack.elasticsearch:9200"
+      ELASTICSEARCH_URL: "http://edx.devstack.elasticsearch7:9200"
```
Comment (stvstnfrd) [inline docker-compose.yml:267]: This is one change that had _not_ already been in the previous WIP PR, but it _seems_ useful?

**Review by stvstnfrd — COMMENTED** [review]
@dianakhuang Who is best to confirm this works for notes?

I'm averse to merging it myself, as I haven't tested notes at all.

File: docker-compose.yml (hunk 1 above)
Comment (kdmccormick) [reply docker-compose.yml:267]: I believe you'll need this; otherwise, notes will keep pointing at the old ES container.

Comment (dianakhuang) [reply docker-compose.yml:267]: It doesn't hurt, but there is a new setting used for ES, which is why I didn't catch it before.

**Comment by dianakhuang** [comment]
I'm having trouble with my docker volumes, but it seems like this works otherwise. I would say we should merge this and https://github.com/edx/edx-notes-api/pull/203 at the same time.

**Review by dianakhuang — APPROVED** [review]

File: docker-compose.yml (hunk 1 above)
Comment (regisb) [reply docker-compose.yml:267]: Unless I'm mistaken, you will need to modify the hardcoded `ELASTICSEARCH_DSL['default']['hosts'] = 'edx.devstack.elasticsearch:9200'` in the devstack settings: https://github.com/edx/edx-notes-api/blob/6e339b1bbbb421efe52cb7737be4b61651a594a4/notesserver/settings/devstack.py#L16

**Review by regisb — CHANGES_REQUESTED** [review]

**Comment by dianakhuang** [comment]
@regisb yup, you are correct. I have a PR up for that: https://github.com/edx/edx-notes-api/pull/203

**Review by davidjoy — APPROVED** [review]

**Review by regisb — CHANGES_REQUESTED** [review]
Would you mind documenting the switch to ES7 in the README? It should include a "Temporary fixes" section which should also describe the progressive switch to MySQL 5.7. This is necessary to inform devstack users of this unexpected behaviour.

EDIT: also, I'd like to see CI run on this PR once https://github.com/edx/edx-notes-api/pull/203 has been merged.

**Comment by dianakhuang** [comment]
We're seeing `Failed to establish a new connection: [Errno 111] Connection refused` after merging the notes PR, so something is still broken in notes. I created https://github.com/edx/devstack/pull/626 to unblock provisioning until we can get ES7 working in devstack here.

EDIT: It is trying to access the correct host: ` http://edx.devstack.elasticsearch7:9200/notes_index`

**Comment by dianakhuang** [comment]
Closing in favor of: https://github.com/edx/devstack/pull/630

**Pull request closed without merging**